icicle/include/icicle/fields/field.h

```diff
@@ -689,14 +689,18 @@ class Field
     return rv;
   }
 
+  inline static std::mt19937 field_rand_generator = std::mt19937{};
```

Comment:
maybe better if you moved all of this to utils and use it here from utils.

Reply:
that's a good idea